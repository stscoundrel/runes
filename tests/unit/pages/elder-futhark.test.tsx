import ReactDOM from 'react-dom'
import ElderFuthark from 'pages/elder-futhark'
import renderer from 'react-test-renderer'

describe('Elder Futhark runic keyboard page', () => {
  test('Does not crash', () => {
    const div = document.createElement('div')
    ReactDOM.render(<ElderFuthark />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  test('Matches snapshot', () => {
    const tree = renderer.create(<ElderFuthark />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
