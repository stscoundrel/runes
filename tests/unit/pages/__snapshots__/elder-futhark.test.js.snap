// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Elder Futhark runic keyboard page Matches snapshot 1`] = `
Array [
  <main
    className="container"
  >
    <section>
      <header>
        <h1>
          Elder Futhark
        </h1>
        <p>
          Translate runes to text
        </p>
      </header>
      <p
        className="result"
      />
      <div
        className="keyboard"
      >
        <button
          aria-label="Runic key ᚠ, meaning letter f"
          className="key"
          onClick={[Function]}
        >
          ᚠ
        </button>
        <button
          aria-label="Runic key ᚢ, meaning letter u"
          className="key"
          onClick={[Function]}
        >
          ᚢ
        </button>
        <button
          aria-label="Runic key ᚦ, meaning letter þ"
          className="key"
          onClick={[Function]}
        >
          ᚦ
        </button>
        <button
          aria-label="Runic key ᚨ, meaning letter a"
          className="key"
          onClick={[Function]}
        >
          ᚨ
        </button>
        <button
          aria-label="Runic key ᚱ, meaning letter r"
          className="key"
          onClick={[Function]}
        >
          ᚱ
        </button>
        <button
          aria-label="Runic key ᚲ, meaning letter k"
          className="key"
          onClick={[Function]}
        >
          ᚲ
        </button>
        <button
          aria-label="Runic key ᚷ, meaning letter g"
          className="key"
          onClick={[Function]}
        >
          ᚷ
        </button>
        <button
          aria-label="Runic key ᚹ, meaning letter w"
          className="key"
          onClick={[Function]}
        >
          ᚹ
        </button>
        <button
          aria-label="Runic key ᚺ, meaning letter h"
          className="key"
          onClick={[Function]}
        >
          ᚺ
        </button>
        <button
          aria-label="Runic key ᚻ, meaning letter h"
          className="key"
          onClick={[Function]}
        >
          ᚻ
        </button>
        <button
          aria-label="Runic key ᚾ, meaning letter n"
          className="key"
          onClick={[Function]}
        >
          ᚾ
        </button>
        <button
          aria-label="Runic key ᛁ, meaning letter i"
          className="key"
          onClick={[Function]}
        >
          ᛁ
        </button>
        <button
          aria-label="Runic key ᛃ, meaning letter j"
          className="key"
          onClick={[Function]}
        >
          ᛃ
        </button>
        <button
          aria-label="Runic key ᛇ, meaning letter ï"
          className="key"
          onClick={[Function]}
        >
          ᛇ
        </button>
        <button
          aria-label="Runic key ᛈ, meaning letter p"
          className="key"
          onClick={[Function]}
        >
          ᛈ
        </button>
        <button
          aria-label="Runic key ᛉ, meaning letter z"
          className="key"
          onClick={[Function]}
        >
          ᛉ
        </button>
        <button
          aria-label="Runic key ᛊ, meaning letter s"
          className="key"
          onClick={[Function]}
        >
          ᛊ
        </button>
        <button
          aria-label="Runic key ᛋ, meaning letter s"
          className="key"
          onClick={[Function]}
        >
          ᛋ
        </button>
        <button
          aria-label="Runic key ᛏ, meaning letter t"
          className="key"
          onClick={[Function]}
        >
          ᛏ
        </button>
        <button
          aria-label="Runic key ᛒ, meaning letter b"
          className="key"
          onClick={[Function]}
        >
          ᛒ
        </button>
        <button
          aria-label="Runic key ᛖ, meaning letter e"
          className="key"
          onClick={[Function]}
        >
          ᛖ
        </button>
        <button
          aria-label="Runic key ᛗ, meaning letter m"
          className="key"
          onClick={[Function]}
        >
          ᛗ
        </button>
        <button
          aria-label="Runic key ᛚ, meaning letter l"
          className="key"
          onClick={[Function]}
        >
          ᛚ
        </button>
        <button
          aria-label="Runic key ᛜ, meaning letter ŋ"
          className="key"
          onClick={[Function]}
        >
          ᛜ
        </button>
        <button
          aria-label="Runic key ᛝ, meaning letter ŋ"
          className="key"
          onClick={[Function]}
        >
          ᛝ
        </button>
        <button
          aria-label="Runic key ᛟ, meaning letter o"
          className="key"
          onClick={[Function]}
        >
          ᛟ
        </button>
        <button
          aria-label="Runic key ᛞ, meaning letter d"
          className="key"
          onClick={[Function]}
        >
          ᛞ
        </button>
        <button
          aria-label="Runic key :, meaning letter  "
          className="key"
          onClick={[Function]}
        >
          :
        </button>
      </div>
    </section>
  </main>,
  <footer
    className="section"
  >
    <div
      className="container text-center"
    >
      <small
        className="copyright"
      >
        Copyright © 
        2021
        <a
          href="https://github.com/stscoundrel"
          rel="noopener noreferrer"
          target="_blank"
        >
          Sampo Silvennoinen / StScoundrel
        </a>
      </small>
    </div>
  </footer>,
]
`;
